apps/designer/app/shared/design-system/components/primitives/dnd/use-move.ts

```diff
@@ -0,0 +1,332 @@
+  shiftKey: boolean;
+  /** Whether the ctrl keyboard modifier was held during the move event. */
+  ctrlKey: boolean;
+  /** Whether the meta keyboard modifier was held during the move event. */
+  metaKey: boolean;
+  /** Whether the alt keyboard modifier was held during the move event. */
+  altKey: boolean;
+}
+
+interface MoveStartEvent extends BaseMoveEvent {
+  /** The type of move event being fired. */
+  type: "movestart";
+  target: HTMLElement;
+  pageX: number;
+  pageY: number;
+}
+
+interface MoveMoveEvent extends BaseMoveEvent {
+  /** The type of move event being fired. */
+  type: "move";
+  /** The amount moved in the X direction since the last event. */
+  deltaX: number;
+  /** The amount moved in the Y direction since the last event. */
+  deltaY: number;
+  pageX: number;
+  pageY: number;
+}
+
+interface MoveEndEvent extends BaseMoveEvent {
+  /** The type of move event being fired. */
+  type: "moveend";
+}
+
+interface MoveEvents {
+  /** Handler that is called when a move interaction starts. */
+  onMoveStart?: (e: MoveStartEvent) => void;
+  /** Handler that is called when the element is moved. */
+  onMove?: (e: MoveMoveEvent) => void;
+  /** Handler that is called when a move interaction ends. */
+  onMoveEnd?: (e: MoveEndEvent) => void;
+}
+
+/**
+ * Handles move interactions across mouse, touch, and keyboard, including dragging with
+ * the mouse or touch, and using the arrow keys. Normalizes behavior across browsers and
+ * platforms, and ignores emulated mouse events on touch devices.
+ */
+export function useMove(props: MoveEvents): MoveResult {
+  let state = useRef<{
+    didMove: boolean;
+    lastPosition: { pageX: number; pageY: number } | null;
+    id: number | null;
+  }>({ didMove: false, lastPosition: null, id: null });
+
+  let { addGlobalListener, removeGlobalListener } = useGlobalListeners();
+
+  // Because addGlobalListener is used to set callbacks,
+  // noramlly it will "see" the version of "props" at the time of addGlobalListener call.
+  // This in turn means that user's callbakcs will see old state variables etc.
+  // To workaround this, we use a ref that always points to the latest props.
+  let latestProps = useRef<MoveEvents>(props);
+  latestProps.current = props;
+
+  let moveProps = useMemo(() => {
+    let moveProps: HTMLAttributes<HTMLElement> = {};
+
+    let start = () => {
+      //disableTextSelection();
+      state.current.didMove = false;
+    };
+    let move = (
+      originalEvent: MouseEvent | TouchEvent | React.KeyboardEvent,
+      pointerType: PointerType,
+      deltaX: number,
+      deltaY: number
+    ) => {
+      if (deltaX === 0 && deltaY === 0) {
+        return;
+      }
+
+      // FIXME: this is a temporary fix
+      // we need to figure out how to properly support keyboard and touch
+      if (
+        !(originalEvent.target instanceof HTMLElement) ||
+        !("pageX" in originalEvent)
+      ) {
+        return;
+      }
```

Comment:
`useMove` uses touch events only as a fallback when Pointer events are not supported. I think we should remove this fallback as it will simplify things for us. https://caniuse.com/pointer

I would also remove the keyboard support for now, but not sure. Do we need to make it work?

Reply:
agree, please make a todo